File: go/LeetCode/decode.go
package main

import "fmt"

func main() {
	encoded := []int{6,5,4,6}
	fmt.Print(decode(encoded))
}

func decode(encoded []int) []int {
	n := len(encoded)
	array := make([]int, n+1)

	arr:=0
	for i:=1;i<=n+1;i++{
		arr^=i
	}
	en:=0
	for i:=1;i<n;i+=2 {
		en^=encoded[i]
	}
	array[0] = arr^en
	for i:=1;i<=n;i++ {
		array[i] = array[i-1]^encoded[i-1]
	}
	return array
}

File: python3/LeetCode/decode.py
from typing import List

class Solution:
    def decode(self, encoded:List[int]) -> List[int]:
        n = len(encoded)
        array = [0] * (n+1)
        arr = 0
        for i in range(1, n+2):
            arr^=i
        en = 0
        for i in range(1, n, 2):
            en^=encoded[i]

        start = arr^en
        array[0] = start
        for i in range(1, n+1):
            array[i] = array[i-1]^encoded[i-1]

        return array



if __name__ == '__main__':
    encoded = [6,5,4,6]
    print(Solution().decode(encoded))